staging/src/k8s.io/client-go/examples/leader-election/main.go

```diff
@@ -131,31 +135,21 @@ func main() {
 			OnStartedLeading: func(ctx context.Context) {
 				// we're notified when we start - this is where you would
 				// usually put your code
-				klog.Infof("%s: leading", id)
+				run(ctx)
 			},
 			OnStoppedLeading: func() {
 				// we can do cleanup here, or after the RunOrDie method
 				// returns
-				klog.Infof("%s: lost", id)
+				klog.Infof("leader lost: %s", id)
```

Comment:
does this still result in the process exiting? if not, I'd take `os.Exit(0)` here

Reply:
done